libraries/Filter/HarmonicNotchFilter.cpp

```diff
@@ -211,6 +229,77 @@ void HarmonicNotchFilter<T>::expand_filter_count(uint16_t total_notches)
     delete[] _old_filters;
 }
 
+/*
+  set the center frequency of a single notch
+ */
+template <class T>
+void HarmonicNotchFilter<T>::set_center_frequency(uint16_t idx, float notch_center, float spread_mul, uint8_t harmonic_mul)
+{
+    const float nyquist_limit = _sample_freq_hz * 0.48f;
+    auto &notch = _filters[idx];
+
+    // scale the notch with the harmonic multiplier
+    notch_center *= harmonic_mul;
+
+    /* disable the filter if its center frequency is above the nyquist
+       frequency
+
+       NOTE: should this be notch_center*spread_mul ? As it is here we
+       do sometimes run the upper notch in a double or triple notch at
+       higher than the nyquist.
+    */
+    if (notch_center >= nyquist_limit) {
+        notch.disable();
```

Comment:
I wonder whether we should consider attenuating as we reach the nyquist limit rather than simply switching off

Reply:
quite likely a good idea, but separate PR I think